### PR TITLE
Rework permission and clientside prop protection

### PR DESCRIPTION
**Pull request by thegrb93**
Still WIP. Post suggestions

### CODE REVIEW

File: lua/starfall/libs_cl/bass.lua [hunk 1]
```diff
@@ -87,7 +87,7 @@ end
 -- @param flags that will control the sound
 -- @param callback to run when the sound is loaded
 function bass_library.loadURL (path, flags, callback)
-	SF.Permissions.check(SF.instance.player, nil, "bass.loadURL")
+	SF.Permissions.check(SF.instance, nil, "bass.loadURL")
```
Comment (adosikas) [inline lua/starfall/libs_cl/bass.lua]: You could expose the path to the permission system here

File: lua/starfall/libs_cl/render.lua [hunk 2]
```diff
@@ -645,14 +645,14 @@ function render_library.getTextureID (tx, cb, alignment, skip_hack)
 	local _1, _2, prefix = tx:find("^(%w-):")
 	if prefix=="http" or prefix=="https" or prefix == "data" then
 		if prefix=="http" or prefix=="https" then
-			SF.Permissions.check(instance.player, nil, "render.urlmaterial")
+			SF.Permissions.check(instance, nil, "render.urlmaterial")
```
Comment (adosikas) [inline lua/starfall/libs_cl/render.lua]: You could expose the URL to the permission system here

File: lua/starfall/libs_sh/builtins.lua [hunk 3]
```diff
@@ -535,7 +535,7 @@ if SERVER then
 	-- @param cmd Command to execute
 	function SF.DefaultEnvironment.concmd (cmd)
 		SF.CheckLuaType(cmd, TYPE_STRING)
-		SF.Permissions.check(SF.instance.player, nil, "console.command")
+		SF.Permissions.check(SF.instance, nil, "console.command")
```
Comment (adosikas) [inline lua/starfall/libs_sh/builtins.lua]: You could expose the command to the permission system here ...

File: lua/starfall/libs_sh/builtins.lua [hunk 4]
```diff
@@ -598,7 +598,7 @@ else
 
 	function SF.DefaultEnvironment.concmd (cmd)
 		SF.CheckLuaType(cmd, TYPE_STRING)
-		SF.Permissions.check(SF.instance.player, nil, "console.command")
+		SF.Permissions.check(SF.instance, nil, "console.command")
```
Comment (adosikas) [inline lua/starfall/libs_sh/builtins.lua]: ... and here

File: lua/starfall/libs_sh/http.lua [hunk 5]
```diff
@@ -51,7 +51,7 @@ end
 -- @param callbackFail the function to be called on request fail, taking the failing reason as an argument
 function http_library.get (url, callbackSuccess, callbackFail)
 	local instance = SF.instance
-	SF.Permissions.check(SF.instance.player, nil, "http.get")
+	SF.Permissions.check(SF.instance, nil, "http.get")
```
Comment (adosikas) [inline lua/starfall/libs_sh/http.lua]: You could expose the URL to the permission system here ...

File: lua/starfall/libs_sh/http.lua [hunk 6]
```diff
@@ -76,7 +76,7 @@ end
 -- @param callbackFail the function to be called on request fail, taking the failing reason as an argument
 function http_library.post (url, params, callbackSuccess, callbackFail)
 	local instance = SF.instance
-	SF.Permissions.check(SF.instance.player, nil, "http.post")
+	SF.Permissions.check(SF.instance, nil, "http.post")
```
Comment (adosikas) [inline lua/starfall/libs_sh/http.lua]: ... and here

File: lua/starfall/libs_sv/prop.lua [hunk 7]
```diff
@@ -67,7 +67,7 @@ end
 -- @return The prop object
 function props_library.create (pos, ang, model, frozen)
 
-	SF.Permissions.check(SF.instance.player,  nil, "prop.create")
+	SF.Permissions.check(SF.instance,  nil, "prop.create")
```
Comment (adosikas) [inline lua/starfall/libs_sv/prop.lua]: You could expose the model to the permission system here

File: lua/starfall/libs_sv/prop.lua [hunk 8]
```diff
@@ -134,7 +134,7 @@ local allowed_components = {
 -- @server
 -- @return Component entity
 function props_library.createComponent (pos, ang, class, model, frozen)
-	SF.Permissions.check(SF.instance.player,  nil, "prop.create")
+	SF.Permissions.check(SF.instance,  nil, "prop.create")
```
Comment (adosikas) [inline lua/starfall/libs_sv/prop.lua]: You could expose the class to the permission system here ...

File: lua/starfall/libs_sv/prop.lua [hunk 9]
```diff
@@ -202,7 +202,7 @@ end
 -- @return The sent object
 function props_library.createSent (pos, ang, class, frozen)
 
-	SF.Permissions.check(SF.instance.player,  nil, "prop.create")
+	SF.Permissions.check(SF.instance,  nil, "prop.create")
```
Comment (adosikas) [inline lua/starfall/libs_sv/prop.lua]: ... and here

File: lua/starfall/libs_cl/bass.lua (hunk 1 above)
Comment (thegrb93) [reply lua/starfall/libs_cl/bass.lua]: Yes someone can do that later if they want to use it for something

**Comment by thegrb93** [comment]
Finished I think.

**Pull request merged**